sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -62,12 +61,24 @@ export class ApplicationController extends BaseController {
     return { data: application.data };
   }
 
+  /**
+   * Submit an existing student application changing the status
+   * to submitted and triggering the necessary processes.
+   * The system will ensure that an application will always be
+   * transitioning from draft to submitted status. The student
+   * application is not supposed to be created directly in the
+   * submitted status in any scenario.
+   * @param payload payload to create the draft application.
+   * @param applicationId application id to be changed to submitted.
+   * @param userToken token from the authenticated student.
+   */
   @AllowAuthorizedParty(AuthorizedParties.student)
-  @Post()
-  async create(
-    @Body() payload: CreateApplicationDto,
+  @Patch(":applicationId/submit")
```

Comment:
Patch with applicationId is expected to update application resource. Is "submit" required here?

Reply:
We also have an @Patch(":applicationId/draft") and the logic for saving and submitting is quite different than I would prefer to have two endpoints instead of just one. Please let me know if it makes sense.

Reply:
Query updated as we talked 😉